core/src/main/java/org/apache/accumulo/core/iterators/PluginEnvironment.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.iterators;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.TableId;
+
+/**
+ * This interface exposes Accumulo system level information to plugins in a stable manner. The
+ * purpose of this interface is to insulate plugins from internal refactorings and changes to
+ * Accumulo.
+ *
+ * @since 2.1.0
+ */
+public interface PluginEnvironment {
+
+  /**
+   * @since 2.1.0
+   */
+  public interface Config extends Iterable<Entry<String,String>> {
```

Comment:
Not sure what to name this.

Reply:
I guess we are running out of different ways to call configuration haha...

Just Configuration like you did in ServiceEnvironment?

Reply:
If ServiceEnvironment is going to extend PluginEnvironment, you could just promote the Configuration interface to PluginEnvironment.

Reply:
I changed the name to Configuration, so its the same in both types.